12-say-grpc/backend/main.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2107 Google Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to writing, software distributed
+// under the License is distributed on a "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os/exec"
+
+	"github.com/Sirupsen/logrus"
+	pb "github.com/campoy/justforfunc/12-say-grpc/api"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+func main() {
+	port := flag.Int("p", 8080, "port to listen to")
+	flag.Parse()
+
+	logrus.Infof("listening to port %d", *port)
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		logrus.Fatalf("could not listen to port %d: %v", *port, err)
+	}
+
+	s := grpc.NewServer()
+	pb.RegisterTextToSpeechServer(s, server{})
+	err = s.Serve(lis)
+	if err != nil {
+		logrus.Fatalf("could not serve: %v", err)
+	}
+}
+
+type server struct{}
+
+func (server) Say(ctx context.Context, text *pb.Text) (*pb.Speech, error) {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, fmt.Errorf("could not create tmp file: %v", err)
+	}
+	if err := f.Close(); err != nil {
+		return nil, fmt.Errorf("could not close %s: %v", f.Name(), err)
+	}
+
+	cmd := exec.Command("flite", "-t", text.Text, "-o", f.Name())
+	if data, err := cmd.CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("flite failed: %s", data)
```

Comment:
I perfectly understand how hard it's to look after every piece of a showcase project. Just wanted to ask, wouldn't it be more accurate to log the actual error here? In some cases, the execution could fail, so `data` might not exist at all, isn't it?